src/parsimmon.js

```diff
@@ -400,7 +400,11 @@ function makeLineColumnIndex(input, i) {
       }
       break;
     }
-    if (input.charAt(j) === "\n") {
+
+    if (
+      input.charAt(j) === "\n" ||
+      (input.charAt(j) === "\r" && input.charAt(j + 1) !== "\n")
```

Comment:
This should probably just check all 3 newline types (https://github.com/jneen/parsimmon/blob/master/API.md#parsimmonnewline)

```js
if (
  input.charAt(j) === "\n" ||
  input.slice(j, j + 2) === "\r\n" ||
  input.charAt(j) === "\r"
)
```

Reply:
I'll check tomorrow but the thinking was to only deal with `"\r"` and let `"\r\n"` parse as it did before, in other words `"\r"` gets skipped over if it is followed by `"\n"` and the next iteration will have `input.charAt(j) === "\n"`.

Does that make sense?

Reply:
Yeah, I just thought this was more direct. If you're counting columns in a text file, you'd never want to count the `\r` in a `\r\n` as a column.

Reply:
Correct and same goes for counting the `"\n"` in `"\r\n"` twice

Once with when `input.charAt(j) === "\n"` evaluates to true
And a second time with `input.slice(j, j + 2) === "\r\n"`

Reply:
I think I'm misunderstanding something, I'll dig back in tomorrow, I think the main thing I am missing is that I thought the scan was forward and now I'm thinking maybe it's backward.

Reply:
Hmm, it's definitely scanning forward. You could do line numbers backwards, but not column numbers.

Reply:
Did you ever get a chance to check back in on my suggestion? If it doesn't work, no big deal, I'm OK with your implementation.

Reply:
> Did you ever get a chance to check back in on my suggestion? If it doesn't work, no big deal, I'm OK with your implementation.

This is a poor showing, sorry. I still haven't come back to touch this PR, I'll let you know as soon as I do, in days and not weeks.

Thank you!

Reply:
No worries!

Reply:
@wavebeem I believe I made the code more readable and by adding a "\r\n" test case I discovered there was an issue with my proposed changes.